components/ILIAS/ResourceStorage/classes/Container/View/ActionBuilder/ExternalActionProvider.php

```diff
@@ -40,14 +40,19 @@ final class ExternalActionProvider implements ActionProvider
      */
     private array $top_actions = [];
 
+    private array $top_actions_modals = [];
+
     public function addSingleAction(string $key, SingleAction $action): void
     {
         $this->single_actions[$key] = $action;
     }
 
-    public function addTopAction(string $key, TopAction $action): void
+    public function addTopAction(string $key, TopAction $action, ?RoundTrip $modal): void
```

Comment:
The new parameter should be `?RoundTrip $modal = null`, I will open a PR with this.